src/main/java/finalRefactor/domain/Ladder.java

```diff
@@ -0,0 +1,42 @@
+package finalRefactor.domain;
+
+import java.util.*;
+
+public class Ladder {
+
+    private final List<LadderLine> lines;
+
+    public Ladder(List<LadderLine> lines) {
+        this.lines = lines;
+    }
+
+    public Ladder(Height height, int countOfPerson) {
+        this(createLines(height, countOfPerson));
+    }
+
+    private static List<LadderLine> createLines(Height height, int countOfPerson) {
+        List<LadderLine> lines = new ArrayList<>();
+        for (int i = 0; i < height.getValue(); i++) {
+            lines.add(new LadderLine(countOfPerson));
+
+        }
+        return lines;
```

Comment:
```suggestion
        return Stream.generate(() -> new LadderLine(countOfPerson))
                .limit(height.getValue())
                .collect(Collectors.toList());
```
스트림을 사용한다면 이렇게 해볼 수 있어요. :)

Reply:
generate.. 감사합니다 하나 더 배웠습니다.
엇 근데 저번에 스트림 적용하라고 하셨던 것은 
```
    public int eachPositionResult(int position) {

        for (LadderLine line : lines) {
            position = line.move(position);
        }
        return position;
    }
```
이코드인대, 이코드에서도 스트림 적용이 되나요??

Reply:
public int eachPositionResult(int position) {
        AtomicInteger result = new AtomicInteger(position);
        
        lines.stream()
                .forEach(line -> result.set(line.move(result.get())));
        
        return result.get();
    }

이렇게 적용해볼 수 있어요 :)